packages/keycloak-user-management/src/forms/index.tsx

```diff
@@ -0,0 +1,295 @@
+import { ErrorMessage, Field, Formik } from 'formik';
+import React, { Dispatch, SetStateAction } from 'react';
+import { Button, Form, notification, Select } from 'antd';
+import * as Yup from 'yup';
+import { history } from '@onaio/connected-reducer-registry';
+import { KeycloakUser } from '@opensrp/store';
+import { KeycloakService } from '@opensrp/keycloak-service';
+import { Dictionary } from '@onaio/utils/dist/types/types';
+
+/** props for editing a user view */
+export interface UserFormProps {
+  accessToken: string;
+  initialValues: KeycloakUser;
+  serviceClass: typeof KeycloakService;
+}
+
+/** interface user action */
+export interface UserAction {
+  alias: string;
+  name: string;
+  providerId: string;
+  enabled: boolean;
+  defaultAction: boolean;
+  priority: number;
+  config: Dictionary;
+}
+
+/** default form initial values */
+export const defaultInitialValues: KeycloakUser = {
+  access: {
+    manageGroupMembership: false,
+    view: false,
+    mapRoles: false,
+    impersonate: false,
+    manage: false,
+  },
+  createdTimestamp: undefined,
+  disableableCredentialTypes: [],
+  email: '',
+  emailVerified: false,
+  enabled: true,
+  firstName: '',
+  id: '',
+  lastName: '',
+  notBefore: 0,
+  requiredActions: [],
+  totp: false,
+  username: '',
+};
+
+/** default props for editing user component */
+export const defaultProps: Partial<UserFormProps> = {
+  accessToken: 'hunter 2',
+  initialValues: defaultInitialValues,
+  serviceClass: KeycloakService,
+};
+
+/** yup validations for practitioner data object from form */
+export const userSchema = Yup.object().shape({
+  lastName: Yup.string().required('Required'),
```

Comment:
Why Yup?

Reply:
Going to be addressed via https://github.com/OpenSRP/web/issues/61